ambry-frontend/src/main/java/com.github.ambry.frontend/GetPeersHandler.java

```diff
@@ -0,0 +1,171 @@
+   * @param restResponseChannel the {@link RestResponseChannel} where headers should be set.
+   * @param callback the {@link Callback} to invoke when the response {@link ReadableStreamChannel} is ready (or if
+   *                 there is an exception).
+   */
+  void handle(RestRequest restRequest, RestResponseChannel restResponseChannel,
+      Callback<ReadableStreamChannel> callback) {
+    RestRequestMetrics requestMetrics =
+        restRequest.getSSLSession() != null ? metrics.getPeersMetrics : metrics.getPeersSSLMetrics;
+    restRequest.getMetricsTracker().injectMetrics(requestMetrics);
+    securityService.processRequest(restRequest,
+        new SecurityProcessRequestCallback(restRequest, restResponseChannel, callback));
+  }
+
+  /**
+   * Callback for the {@link SecurityService} that handles generating the replica list if the security checks
+   * succeeded.
+   */
+  private class SecurityProcessRequestCallback implements Callback<Void> {
+    private final RestRequest restRequest;
+    private final RestResponseChannel restResponseChannel;
+    private final Callback<ReadableStreamChannel> callback;
+    private final long operationStartTimeMs;
+
+    SecurityProcessRequestCallback(RestRequest restRequest, RestResponseChannel restResponseChannel,
+        Callback<ReadableStreamChannel> callback) {
+      this.restRequest = restRequest;
+      this.restResponseChannel = restResponseChannel;
+      this.callback = callback;
+      operationStartTimeMs = SystemTime.getInstance().milliseconds();
+    }
+
+    /**
+     * If {@code exception} is null, gathers all the peers of the datanode that corresponds to the params in the request
+     * and returns them as a part of the response body.
+     * @param result The result of the request. This would be non null when the request executed successfully
+     * @param exception The exception that was reported on execution of the request
+     */
+    @Override
+    public void onCompletion(Void result, Exception exception) {
+      long processingStartTimeMs = SystemTime.getInstance().milliseconds();
+      metrics.getPeersSecurityProcessingTimeInMs.update(processingStartTimeMs - operationStartTimeMs);
+      try {
+        if (exception == null) {
+          DataNodeId dataNodeId = getDataNodeId(restRequest);
+          LOGGER.trace("Getting peer hosts for {}:{}", dataNodeId.getHostname(), dataNodeId.getPort());
+          Set<DataNodeId> peerDataNodeIds = new HashSet<>();
+          List<? extends ReplicaId> replicaIdList = clusterMap.getReplicaIds(dataNodeId);
+          for (ReplicaId replicaId : replicaIdList) {
+            List<? extends ReplicaId> peerReplicaIds = replicaId.getPeerReplicaIds();
+            for (ReplicaId peerReplicaId : peerReplicaIds) {
+              peerDataNodeIds.add(peerReplicaId.getDataNodeId());
+            }
+          }
+          StringBuilder peerSb = new StringBuilder();
```

Comment:
Why did we decide to go with a comma separated list instead of a JSON object for this response like in GET replicas in ambry-admin?